src/app/app.component.ts

```diff
@@ -153,33 +193,66 @@ export class AppComponent implements OnInit {
   }
 
   onDrop(ev: DragEvent) {
-    ev.preventDefault();
     this.dropzoneHighlight = false;
 
-    const dropFile = (file: File) => {
-      if (/\.(?:htmlz|epub)$/.test(file.name)) {
-        this.onFileChange(file);
-      } else {
-        alert('Only HTMLZ and EPUB files are supported');
+    if (!ev?.dataTransfer?.items) {
+      return;
+    }
+
+    const items = [];
+
+    for (const item of ev.dataTransfer.items) {
+      if ('file' !== item.kind) {
+        continue;
       }
-    };
 
-    if (ev.dataTransfer) {
-      if (ev.dataTransfer.items) {
-        const item = ev.dataTransfer.items[0];
-        if (item) {
-          const file = item.getAsFile();
-          if (file) {
-            dropFile(file);
-          }
+      items.push(item.webkitGetAsEntry());
```

Comment:
Are there no better alternatives than making use of "webkit" prefixed functions?

Reply:
only native filesystem comes to my mind but that one is really new and only available in chromium browsers. Also no other methods listed on MDN and in the wc3 examples they are also using this one. Firefox uses this method like for 40+ versions already and even safari implemented it n 11.3.. so for now relative safe to use i guess